### DISCUSSION TITLE
More flexible recurring events beyond weekly recurrence

### DISCUSSION THREAD

**Discussion by menagerie198** [davish/obsidian-full-calendar#344] [Feature Requests]
Could we have the ability to set recurrence more finely. Currently I can only get weekly recurrence to work. Monthly would be great More fine grained recurrence - fortnightly, every third Tuesday, etc would be gravy.

**Comment by davish** [comment]
FullCalendar has support for [`rrule`](https://fullcalendar.io/docs/rrule-plugin) that we could adopt. The tricky part will be figuring out a good UI -- might be good to crib off of GCal for their advanced recurring options?

**Comment by menagerie198** [comment]
That would be great. MS Outlook on line is also very slick in this regard, slightly more intuitive than Gcal, I think.

**Comment by davish** [comment]
Could you share a screenshot? I haven't used outlook calendar too much before.

**Comment by menagerie198** [comment]
Sure... 
![image](https://user-images.githubusercontent.com/7330436/155510504-2ea2a28d-a1ad-4450-bd3f-0435226539c6.png)
![image](https://user-images.githubusercontent.com/7330436/155510593-b385ee7e-4cae-41c7-b7d7-8ea04d2f2e6f.png)

**Comment by menagerie198** [comment]
![image](https://user-images.githubusercontent.com/7330436/155510875-c16e8b53-657b-4330-9589-b6d6c00ca2e4.png)

**Comment by lumargh** [comment]
Slated in the Obsidian store has a great recurrence UI and functionality. I'll agree with the majority that an expanded recurrence functionality in Full Cal would be great.
<img width="542" alt="image" src="https://user-images.githubusercontent.com/106437649/184627666-4fef5de0-dd5c-498b-aa9a-b42fd2852676.png">

**Comment by SirSceptalot** [comment]
Looking for this as well.
Using Obsidian for IT documentation, and many recurring tasks such as data integrity checks, backup etc. have multi-week or multi-month periods.

**Comment by macleod-ee** [comment]
Here is some more UI inspiration, coming from Thunderbird. 

Simpler for most common usage:
![image](https://user-images.githubusercontent.com/4322153/203124002-2a3e7ad0-2107-4b98-a98d-f764ac20c708.png)

More complex custom usage, this is probably not a priority for some time of course.
![image](https://user-images.githubusercontent.com/4322153/203124142-fe766b5e-0386-4412-9bbf-a17349869f16.png)

**Comment by macleod-ee** [comment]
My current workaround is to set all non-weekly recurring events (for IT work, bills, etc) as a private ics calendar in Zoho Calendar (or any other calendar service). This then let's it show up on the FC calendar (close the calendar tab and reopen on first load).

**Comment by Jethril** [comment]
I would suggest the Linux crontab syntax to give the users a chance to input a rule that would not be available in the UI. It could also be used to save (serialize) the rule to a text file.

**Comment by davish** [comment]
Wow, @Jethril that is a great idea :)

**Reply by FiniTea** [reply]
Is there any update on this, is a feature like this on the roadmap? Yearly recurring intervals would be really useful!

**Comment by Jethril** [comment]
I would like to give it a try, but I must learn how the plugins are made (seems very simple).

**Comment by reddwest** [comment]
I don't currently have any suggestions but I've been driving myself crazy trying to figure out a workaround to do yearly events like birthdays, and would definitely appreciate further options/flexibility for this

**Reply by fatherofinvention** [reply]
Only thing I can think of is to offload the missing functionality to a user created script (if you know how to write Python, JS, PowerShell, etc.) using other plugins like QuickAdd and/or Shell Commands. Your script would generate the markdown files with the appropriate frontmatter which would then be usable by the FullCalendar plugin.

**Comment by crux-03** [comment]
If the UI is still up for discussion, would something like this be adequate?

![image](https://github.com/davish/obsidian-full-calendar/assets/25249091/2aaa1a90-826a-472f-9cae-4d14d4ae0d50)

**Comment by bskeen** [comment]
Hey everyone, I just added a PR for a first attempt at this functionality: #500 
Please take a look and let me know what you think. I am definitely open to adjusting the UI/logic/supported recurrence types if needed.

Here are some screenshots:
![image](https://github.com/davish/obsidian-full-calendar/assets/9536373/ad7070e1-a460-42e0-a362-c8b7e1b8bce1)

![image](https://github.com/davish/obsidian-full-calendar/assets/9536373/392155c1-02a5-4024-9474-abe864bec5cc)
![image](https://github.com/davish/obsidian-full-calendar/assets/9536373/96dd06f9-7cdc-47e7-957a-a80ab9ca965a)
![image](https://github.com/davish/obsidian-full-calendar/assets/9536373/6372f91e-2740-4002-956c-be1ea2876525)

**Comment by manveldavid** [comment]
Can you give an example of a flexible recurring event please? What syntax should I use? Nothing about this in the documentation or on YouTube...

**Reply by manveldavid** [reply]
add some .md file please

**Comment by manveldavid** [comment]
Omg, I found something in sources
`example.md`
```
---
type: rrule
allDay: false
startDate: 2022-03-01
endTime: 12:30
startTime: 11:00
rrule: FREQ=WEEKLY;BYDAY=SU,MO,TU,WE,TH,FR,SA;INTERVAL=2
skipDates: []
title: Hello world!
---
```

**Comment by mjluser1** [comment]
I just started using the calendar this week and love it---except for my annual, bimonthly, monthly, and biweekly events. Thanks for working on this!!

**Comment by mjluser1** [comment]
@fatherofinvention Is there a way to get the calendar to show the events if the markdown files are created for the events? I just duplicated a file and changed the dates, but I do not know how to get the calendar to see it.

**Reply by manveldavid** [reply]
Open your note in source mode (</> icon), then check 'skipDates: []'

**Comment by YouFoundJK** [comment]
This should now be possible in my fork [here](https://github.com/YouFoundJK/plugin-full-calendar/releases/latest), if anyone is interested!
